### PR TITLE
gh-90699: Use _Py_STR(empty) instead of PyUnicode_New(0, 0)

### PR DESCRIPTION
**Pull request by sweeneyde**

### CODE REVIEW

**Review by KarenZhuang — APPROVED** [review]

File: Modules/_ctypes/_ctypes.c [hunk 1]
```diff
@@ -4635,7 +4635,7 @@ Array_subscript(PyObject *myself, PyObject *item)
             wchar_t *dest;
 
             if (slicelen <= 0)
-                return PyUnicode_New(0, 0);
+                return Py_NewRef(&_Py_STR(empty));
```
Comment (corona10) [inline Modules/_ctypes/_ctypes.c]: Out of curiosity, what is the benefit of this change except inline calling?
AFAIK, `PyUnicode_New(0, 0);` will finally call `unicode_get_empty` which is the same replacement of `_Py_STR(empty)`

https://github.com/python/cpython/blob/f33e2c87a83917b5139d97fd8ef7cba7223ebef5/Objects/unicodeobject.c#L1372-L1377

https://github.com/python/cpython/blob/f33e2c87a83917b5139d97fd8ef7cba7223ebef5/Objects/unicodeobject.c#L259-L271

Comment (corona10) [reply Modules/_ctypes/_ctypes.c]: AFAIK, @ericsnowcurrently try to reduce private API from the stdlib module as the best practice for 3rd party users who try to write C extension module by referencing stdlib implementation..
So I don't think this is a good replacement if there are no significant improvements.

Please let me know if I know wrong Eric :)

Comment (sweeneyde) [reply Modules/_ctypes/_ctypes.c]: That makes sense. I had first seen that ceval.c could change, so I made similar changes elsewhere, but if the other code is better left alone, I don't mind.

I updated this PR to only have the ceval.c change, where the code can actually get smaller and have fewer branches.

**Review by corona10 — APPROVED** [review]
LGTM
I am +1 with an aggressive optimization approach for opcode evaluation :)

**Pull request merged**